stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/MessagesScreen.kt

```diff
@@ -170,17 +170,38 @@ public fun MessagesScreen(
             )
         }
 
-        if (selectedMessage != null) {
-            SelectedMessageOverlay(
-                messageOptions = defaultMessageOptionsState(selectedMessage, user, listViewModel.isInThread),
-                message = selectedMessage,
-                currentUser = user,
-                onMessageAction = { action ->
-                    composerViewModel.performMessageAction(action)
-                    listViewModel.performMessageAction(action)
-                },
-                onDismiss = { listViewModel.removeOverlay() }
-            )
+        AnimatedVisibility(
+            visible = selectedMessage != null,
+            enter = fadeIn(),
+            exit = fadeOut(animationSpec = tween(delayMillis = AnimationConstants.DefaultDurationMillis / 2))
+        ) {
+            if (selectedMessage != null) {
```

Comment:
This animation doesn't work correctly when exiting, because as soon as the selectedMessage is null, we remove it from the UI and it just blinks.

Maybe we can change the condition to:

```
val currentlySelectedMessage = selectedMessage ?: Message()

AnimatedVisibility(
            visible = currentlySelectedMessage .id.isNotEmpty(),
```

This fixes the behavior

Reply:
Very good suggestion.

This is something that I've noticed before but could not work around so easily. 

I've refactored it as per the suggestion.